### ISSUE TITLE
disable geo lat long to US address reverse lookup

### ISSUE THREAD

**Issue by meonkeys** [gino0631/nextcloud-metadata#84]
I noticed metadata for some of my photos includes a "Location" line with a human-readable United States address. How is this geo-to-location translation performed?

Ah, looks like [Nominatim / OSM](https://github.com/gino0631/nextcloud-metadata/blob/a8d69c8337055908ac62289ffcaaf86d6f830c3f/js/tabview.js#L88). Is that right? Is there any way to turn this off? I guess I'd rather not reach out to that service because I don't need the "Location" line in the metadata.

**Comment by gino0631** [comment]
Yes, Nominatim. No, location is resolved for all picture files that have coordinates (regardless of the country), and there are currently no options to disable this behaviour (except for blocking access to that service at the server level).

**Comment by meonkeys** [comment]
Thank you.

I forgot about filtering traffic to the domain altogether... I'll do that with my Pi-hole.

**Comment by meonkeys** [comment]
Ok, I tried it, and unfortunately a Nextcloud notification pops up every time it tries reverse geocoding, saying: "Connection to server lost". Just a bit annoying, but it does of course prevent any connections to Nominatim since the DNS lookup fails (since the Pi-hole returns `0.0.0.0`).

**Comment by meonkeys** [comment]
See also: #85

**Comment by gino0631** [comment]
Maybe it is possible to implement a better workaround, but in any case it would be just a workaround.

I haven't considered that the use of Nominatim could have an impact on privacy, but I'm open to argument.

To understand better what was already said - the issue mentions "United States address" - is there anything special about that country addresses?

As this can potentially be a feature, I'm reopening.

**Comment by meonkeys** [comment]
> I haven't considered that the use of Nominatim could have an impact on privacy, but I'm open to argument.

I'm sure they're fine, just allow me to play devil's advocate.

I don't know anything about the service. Someone runs their servers, right? Who are they? Where are they, and what laws are they governed by? What data are they sent (IP addresses for my request, lat/long coords for my photos, etc)? Who has usage rights for these data besides Nominatim? What does Nominatim do with the data? What are their terms? Did I agree to their EULA? What rights do I have over the data I send? What about users in Europe or California?

Again, I'm sure it's all fine, this code is fine, they're fine, etc... but I'd rather not care to even think about Nominatim and I don't want or need the reverse geocoding feature, so I'd rather not send any data their way at all. I try to generally follow a "local first" principle when it comes to my data. This includes reverse geocoding, face recognition, stuff like that. If I can do it offline, I prefer to. I thought given the "private cloud" ethos of Nextcloud that other admins/users might appreciate having this manner of control as well.

> the issue mentions "United States address" - is there anything special about that country addresses?

Nope, that's just where I live, so I wasn't sure how it behaves for addresses outside the US.

> As this can potentially be a feature, I'm reopening.

Cool, thanks. That was my idea as well, that this could be a feature/enhancement if it is desirable. If not, like you said: there are extant workarounds or we could figure out others.

Thanks for your time, -Adam
